### ISSUE TITLE
Multiseat and IPC

### ISSUE THREAD

**Issue by acrisci** [swaywm/sway#1588]
The focus model for wlroots makes things a bit more complicated to accommodate multiple seats. We can no longer assume that there is a single focused container, but rather each seat has its own focus state for the whole tree.

The problem is that to report this info breaks every ipc library that operates on this assumption.

We discussed having different types when sway is configured with a single seat than if it is multiseat, but we wantto avoid variant types because they make ipc libraries difficult to use and write.

One solution would be for the `get_tree` command to take a seat parameter and report focus for only that seat (similar for how we plan to do it for commands). Example:

```
swaymsg -t get_tree --seat=seat0
```

Another solution might be to change how focus is reported from a boolean to a list of seat names, with an empty list meaning it is not focused on any seat.

It's difficult to know what the best solution is because nobody has come up with applications for multiseat ipc.

There are probably other good solutions people can think of too.

**Comment by David96** [comment]
Not sure whether this is related, but I just tried to adapt the transparency script for a multiseat environment with multiple focused windows and had to find out that the tree returned from sway ipc always only has one focused window in it.
Do I understand that right that this is intentional so far until we have a solution which returns the focus states for different seats separately? Because intuitively I would have expected the tree to simply include multiple focused containers in a multiseat setup.
The current state makes supporting multiple seats with the transparency script impossible as I see it.
If it's not related, I'll create a new issue for it, just thought I'd ask first.

**Comment by emersion** [comment]
>had to find out that the tree returned from sway ipc always only has one focused window in it.

Yes, this is related and this is the issue discussed here. No, we haven't decided how to fix it yet. Some ideas are in the first post.
